src/v0/destinations/facebook_conversions/utils.js

```diff
@@ -0,0 +1,204 @@
+  }
+
+  return category;
+};
+
+const populateContentsAndContentIDs = (productPropertiesArray, fallbackQuantity) => {
+  const contentIds = [];
+  const contents = [];
+  if (Array.isArray(productPropertiesArray)) {
+    productPropertiesArray.forEach((productProps) => {
+      if (isObject(productProps)) {
+        const productId = productProps.product_id || productProps.sku || productProps.id;
+        if (productId) {
+          contentIds.push(productId);
+          contents.push({
+            id: productId,
+            quantity: productProps.quantity || fallbackQuantity || 1,
+            item_price: productProps.price,
+          });
+        }
+      }
+    });
+  }
+
+  return { contentIds, contents };
+};
+
+const populateCustomDataBasedOnCategory = (customData, message, category, categoryToContent) => {
+  let eventTypeCustomData = {};
+  if (category.name) {
+    eventTypeCustomData = constructPayload(message, MAPPING_CONFIG[category.name]);
+  }
+
+  switch (category.type) {
+    case 'product list viewed': {
+      const { contentIds, contents } = populateContentsAndContentIDs(
+        message.properties?.products,
+        message.properties?.quantity,
+      );
+
+      const contentCategory = eventTypeCustomData.content_category;
+      let contentType;
+      if (contentIds.length > 0) {
+        contentType = 'product';
+      } else if (contentCategory) {
+        contentIds.push(contentCategory);
+        contents.push({
+          id: contentCategory,
+          quantity: 1,
+        });
+        contentType = 'product_group';
+      }
+
+      eventTypeCustomData = {
+        ...eventTypeCustomData,
+        content_ids: contentIds,
+        contents,
+        content_type: getContentType(
+          message,
+          contentType,
+          categoryToContent,
+          DESTINATION.toLowerCase(),
+        ),
+        content_category: getContentCategory(contentCategory),
+      };
+      break;
+    }
+    case 'product added':
+    case 'product viewed':
+    case 'products searched':
+    case 'payment info entered':
+    case 'product added to wishlist': {
+      const contentCategory = eventTypeCustomData.content_category;
+      const contentType = eventTypeCustomData.content_type;
+      const { contentIds, contents } = populateContentsAndContentIDs([message.properties]);
+      eventTypeCustomData = {
+        ...eventTypeCustomData,
+        content_ids: contentIds,
+        contents,
+        content_type: getContentType(
+          message,
+          contentType,
+          categoryToContent,
+          DESTINATION.toLowerCase(),
+        ),
+        content_category: getContentCategory(contentCategory),
+      };
+
+      const query = eventTypeCustomData.search_string;
+      const validQueryType = ['string', 'number', 'boolean'];
+      if (query && !validQueryType.includes(typeof query)) {
+        throw new InstrumentationError("'query' should be in string format only");
+      }
+
```

Comment:
It is related to product searched event only, So can you move it to some function related to with name something like validatProductSearched ?

Reply:
Done